crates/adapters/src/transport/url.rs

```diff
@@ -0,0 +1,248 @@
+        let config = UrlInputConfig::deserialize(config)?;
+        let ep = UrlInputEndpoint::new(config);
+        Ok(Box::new(ep))
+    }
+}
+
+/// Configuration for reading data from an HTTP or HTTPS URL with
+/// [`UrlInputTransport`].
+#[derive(Clone, Deserialize, ToSchema)]
+pub struct UrlInputConfig {
+    /// URL.
+    pub path: String,
+}
+
+struct UrlInputEndpoint {
+    config: UrlInputConfig,
+    sender: Sender<PipelineState>,
+    receiver: Receiver<PipelineState>,
+}
+
+impl UrlInputEndpoint {
+    fn new(config: UrlInputConfig) -> Self {
+        let (sender, receiver) = channel(PipelineState::Paused);
+        Self {
+            config,
+            sender,
+            receiver,
+        }
+    }
+
+    async fn worker_thread(
+        config: UrlInputConfig,
+        consumer: &mut Box<dyn InputConsumer>,
+        mut receiver: Receiver<PipelineState>,
+    ) -> AnyResult<()> {
+        let client = Client::builder()
+            .connector(Connector::new().rustls(rustls_config()))
+            .finish();
+        let request = client.get(config.path);
+        let mut response = request.send().await.map_err(
+            // `awc` intentionally uses errors that aren't `Sync`, but
+            // `anyhow::Error` requires `Sync`.  Transform the error so we can
+            // return it.
+            |error| anyhow!("{error}"),
+        )?;
+
+        while *receiver.borrow() != PipelineState::Terminated {
+            select! {
+                _ = receiver.changed() => (),
+                result = response.next(), if *receiver.borrow() == PipelineState::Running => {
+                    match result {
+                        None => return Ok(()),
+                        Some(Ok(data)) => consumer.input_fragment(&data)?,
+                        Some(Err(error)) => Err(error)?,
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl InputEndpoint for UrlInputEndpoint {
+    fn connect(&mut self, mut consumer: Box<dyn InputConsumer>) -> AnyResult<()> {
+        let config = self.config.clone();
+        let receiver = self.receiver.clone();
+        let _worker = spawn(move || {
+            System::new().block_on(async move {
+                let result = Self::worker_thread(config, &mut consumer, receiver).await;
+                if let Some(error) = result.err().or_else(|| consumer.eoi().err()) {
+                    consumer.error(true, error);
+                }
+            });
+        });
+        Ok(())
+    }
+
+    fn pause(&self) -> AnyResult<()> {
+        Ok(self.sender.send(PipelineState::Paused)?)
+    }
+
+    fn start(&self) -> AnyResult<()> {
+        Ok(self.sender.send(PipelineState::Running)?)
+    }
+
+    fn disconnect(&self) {
+        let _ = self.sender.send(PipelineState::Terminated);
+    }
+}
+
+impl Drop for UrlInputEndpoint {
+    fn drop(&mut self) {
+        self.disconnect();
+    }
+}
+
+fn rustls_config() -> Arc<ClientConfig> {
```

Comment:
what does this name stand for?

Reply:
shouldn't the name be rusttls?

Reply:
> shouldn't the name be rusttls?

Feel free to file a bug upstream: https://github.com/rustls/rustls/issues